File: PayHeureUI/src/app/services/paie.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PaieCalculRequest, PaieCalculResponse } from '../models/PaieCalcul';
import { API_BASE_URL } from '../config/api.config';

@Injectable({
  providedIn: 'root'
})
export class PaieService {

  private readonly apiUrl = `${API_BASE_URL}/paie`;

  constructor(private http: HttpClient) {}

  /**
   * Récupère les pointages du salarié sur la période demandée et calcule le montant dû, au
   * taux horaire fourni (jamais transmis ailleurs qu'à cet appel).
   */
  calculer(request: PaieCalculRequest): Observable<PaieCalculResponse> {
    return this.http.post<PaieCalculResponse>(`${this.apiUrl}/calcul`, request);
  }
}

File: src/main/java/com/example/payheurebackend/service/PaieService.java
package com.example.payheurebackend.service;

import com.example.payheurebackend.dto.PaieCalculRequest;
import com.example.payheurebackend.dto.PaieCalculResponse;

/** Calcul de la paie d'un salarié à partir de ses pointages, sur une période donnée. */
public interface PaieService {

    /**
     * Récupère les pointages du salarié sur la période demandée, reconstitue les sessions de
     * travail (entrée/sortie) et calcule le montant dû au taux horaire fourni.
     *
     * @throws com.example.payheurebackend.exception.ResourceNotFoundException si le salarié n'existe pas
     * @throws com.example.payheurebackend.exception.InvalidPeriodException si la période est incohérente
     */
    PaieCalculResponse calculer(PaieCalculRequest request);
}
